src/foam/lib/NetworkPropertyPredicate.js

```diff
@@ -0,0 +1,35 @@
+foam.CLASS({
+    package: 'foam.lib',
+    name: 'NetworkPropertyPredicate',
+    extends: 'foam.lib.PermissionedPropertyPredicate',
+    javaImports: [
+        'foam.nanos.auth.AuthService',
+        'foam.lib.PermissionedPropertyPredicate'
+    ],
+    
+    methods: [
+      {
+        name: 'propertyPredicateCheck',
```

Comment:
Since this is part of the interface, you don't need the type, or args. They're pulled in automatically.